### PR TITLE
CUMULUS-1029 add glossary

### PR DESCRIPTION
**Pull request by Jkovarik**
**Summary:** Summary of changes

Addresses [CUMULUS-1029: Glossary of doc terms](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1029)

## Changes

* Adds glossary to Cumulus docs

To review, run the local docs server and look at http://localhost:3000/cumulus/docs/next/glossary

### CODE REVIEW

File: docs/glossary.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
+---
```
Comment (laurenfrederick) [inline docs/glossary.md]: Some that I think are missing:
- granule
- CNM
- CMR
- DAAC

File: docs/glossary.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+# AWS Glossary
+
+For terms/items from Amazon/AWS not mentioned in this glossary, please refer to the [AWS Glossary](https://docs.aws.amazon.com/general/latest/gr/glos-chap.html).
+
+# Cumulus Glossary of Terms
+
+### API Gateway
+
+  A Cumulus component that provides an API that provides accessors to Cumulus functionality.
```
Comment (laurenfrederick) [inline docs/glossary.md]: API Gateway is really an AWS thing. Cumulus uses API gateway for its API. https://aws.amazon.com/api-gateway/

File: docs/glossary.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+  This component is published as a [NPM package](https://www.npmjs.com/package/@cumulus/apiAPI).   API documentation can be viewed [here](https://nasa.github.io/cumulus-api/).
```
Comment (laurenfrederick) [inline docs/glossary.md]: This link is incorrect. Should be https://www.npmjs.com/package/@cumulus/api

File: docs/glossary.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+### ARN
+
+  Refers to an AWS "Amazon Resource Name".
+
+  For more info, see the [AWS documentation](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html).
+
+### AWS
+
+  Amazon's Cloud Services Platform "Amazon Web Services".
```
Comment (laurenfrederick) [inline docs/glossary.md]: I would just link to aws.amazon.com

File: docs/glossary.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+
+  A library designed to help task developers integrate step function tasks into a Cumulus workflow by adapting task input/output into the Cumulus Message format.
+
+  For more information, see [CMA workflow reference page](workflows/input_output#cumulus-message-adapter).
+
+### Dead Letter Queue (DLQ)
+
+  This refers to Amazon SQS Dead-Letter Queues - these SQS queues are specifically configured to capture failed messages from other services/SQS queues/etc to allow for processing of failed messages.
+
+  For more on DLQs, see the [Amazon Documentation](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-dead-letter-queues.html) and the [Cumulus DLQ feature page](features/dead_letter_queues.md).
```
Comment (laurenfrederick) [inline docs/glossary.md]: DLQ link did not work

File: docs/glossary.md (hunk 2 above)
Comment (Jkovarik) [reply docs/glossary.md]: 🤔  Right, in the context of this doc I believe there was at least one reference to the Cumulus API Gateway that seemed to overload the term.   In either case we should probably re-write this a bit to highlight that 👍

File: docs/glossary.md (hunk 1 above)
Comment (Jkovarik) [reply docs/glossary.md]: 👍  on the last three.        Granule could be a pretty complex definition 🤔

File: docs/glossary.md (hunk 2 above)
Comment (Jkovarik) [reply docs/glossary.md]: 03e1647c

File: docs/glossary.md (hunk 3 above)
Comment (Jkovarik) [reply docs/glossary.md]: 03e1647c

File: docs/glossary.md (hunk 1 above)
Comment (Jkovarik) [reply docs/glossary.md]: ffc8d38d, 7ea49dea.   We don't yet have a link to the CNM standard.

File: docs/glossary.md (hunk 4 above)
Comment (Jkovarik) [reply docs/glossary.md]: 03e1647

**Comment by Jkovarik** [comment]
@laurenfrederick  I believe your concerns have been addressed

**Pull request merged**